internal/attestation/crafter/materials/helmchart.go

```diff
@@ -0,0 +1,128 @@
+
+	schemaapi "github.com/chainloop-dev/chainloop/app/controlplane/api/workflowcontract/v1"
+	api "github.com/chainloop-dev/chainloop/internal/attestation/crafter/api/attestation/v1"
+	"github.com/chainloop-dev/chainloop/internal/casclient"
+	"github.com/rs/zerolog"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	ChartFileName      = "Chart.yaml"
+	ValuesYamlFileName = "values.yaml"
+)
+
+type HelmChartCrafter struct {
+	backend *casclient.CASBackend
+	*crafterCommon
+}
+
+func NewHelmChartCrafter(materialSchema *schemaapi.CraftingSchema_Material, backend *casclient.CASBackend,
+	l *zerolog.Logger) (*HelmChartCrafter, error) {
+	if materialSchema.Type != schemaapi.CraftingSchema_Material_HELM_CHART {
+		return nil, fmt.Errorf("material type is not HELM_CHART format")
+	}
+
+	return &HelmChartCrafter{
+		backend:       backend,
+		crafterCommon: &crafterCommon{logger: l, input: materialSchema},
+	}, nil
+}
+
+func (c *HelmChartCrafter) Craft(ctx context.Context, filepath string) (*api.Attestation_Material, error) {
+	// Open the helm chart tar file
+	f, err := os.Open(filepath)
+	if err != nil {
+		return nil, fmt.Errorf("can't open the file: %w", err)
+	}
+	defer f.Close()
+
+	// Decompress the file if possible
+	uncompressedStream, err := gzip.NewReader(f)
+	if err != nil {
+		return nil, fmt.Errorf("can't uncompress file, unexpected material type: %w", err)
+	}
+
+	// Create a tar reader
+	tarReader := tar.NewReader(uncompressedStream)
+
+	// Flags to track whether required files are found
+	chartFileValid, chartValuesValid := false, false
+
+	// Iterate through the files in the tar archive
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			// Reached the end of tar archive
+			break
+		}
+		if err != nil {
+			return nil, fmt.Errorf("error reading tar file: %w", err)
+		}
+
+		// Check if the file is a regular file
+		if header.Typeflag != tar.TypeReg {
+			continue // Skip if it's not a regular file
+		}
+
+		// Validate Chart.yaml and values.yaml files
+		if strings.Contains(header.Name, ChartFileName) {
+			if err := validateYamlFile(tarReader); err != nil {
+				return nil, fmt.Errorf("invalid Chart.yaml file: %w", err)
+			}
+			chartFileValid = true
+		} else if strings.Contains(header.Name, ValuesYamlFileName) {
+			if err := validateYamlFile(tarReader); err != nil {
+				return nil, fmt.Errorf("invalid values.yaml file: %w", err)
+			}
+			chartValuesValid = true
+		}
+
+		// Stop iterating if both files are found
+		if chartValuesValid && chartFileValid {
+			break
+		}
+	}
+
+	// If the chart.yaml and values.yaml files are not found, return an error
+	if !chartFileValid || !chartValuesValid {
+		return nil, fmt.Errorf("missing required files in the helm chart: Chart.yaml and values.yaml")
+	}
+
+	// Upload and craft the chart
+	return uploadAndCraft(ctx, c.input, c.backend, filepath, c.logger)
+}
+
+// validateYamlFile validates the YAML file just by trying to unmarshal it
+func validateYamlFile(r io.Reader) error {
```

Comment:
Please add a receiver to this method so that it's not exposed at package level (unless we want to expose it for some reason).

Reply:
Hmm no no, we don't need to expose it. Eventually in the future this can belong to a shared/utils part to be reused in the the code. Right now, there is no need for that neither for it be exposed.